modules/tinymce/src/models/dom/main/ts/table/api/Options.ts

```diff
@@ -14,21 +14,42 @@ const option: {
 } = (name: string) => (editor: Editor) =>
   editor.options.get(name);
 
+// Note: This is also contained in the table plugin Options.ts file
+const stylesAvailableAsAttributes = [ 'width' ];
+
+const getPixelForcedWidth = (editor: Editor) => {
+  // Determine the inner size of the parent block element where the table will be inserted
+  const dom = editor.dom;
+  const parentBlock = dom.getParent<HTMLElement>(editor.selection.getStart(), dom.isBlock) ?? editor.getBody();
+  return Width.getInner(SugarElement.fromDom(parentBlock)) + 'px';
+};
+
 // Note: This is also contained in the table plugin Options.ts file
 const determineDefaultTableStyles = (editor: Editor, defaultStyles: Record<string, string>): Record<string, string> => {
-  if (isTablePixelsForced(editor)) {
-    // Determine the inner size of the parent block element where the table will be inserted
-    const dom = editor.dom;
-    const parentBlock = dom.getParent<HTMLElement>(editor.selection.getStart(), dom.isBlock) ?? editor.getBody();
-    const contentWidth = Width.getInner(SugarElement.fromDom(parentBlock));
-    return { ...defaultStyles, width: contentWidth + 'px' };
-  } else if (isTableResponsiveForced(editor)) {
-    return Obj.filter(defaultStyles, (_value, key) => key !== 'width');
+  if (isTableResponsiveForced(editor) || shouldStyleWithCss(editor) === false) {
+    return Obj.filter(defaultStyles, (_value, key) => !Arr.contains(stylesAvailableAsAttributes, key));
+  } else if (isTablePixelsForced(editor)) {
+    return { ...defaultStyles, width: getPixelForcedWidth(editor) };
   } else {
     return defaultStyles;
   }
 };
 
+// Note: This is also contained in the table plugin Options.ts file
+const determineDefaultTableAttributes = (editor: Editor, defaultAttributes: Record<string, string>): Record<string, string> => {
+  const defaultStyles = editor.options.get('table_default_styles');
+
+  if (editor.options.isSet('table_default_styles')) {
+    return defaultAttributes;
+  } else if (shouldStyleWithCss(editor)) {
+    return defaultAttributes;
+  } else if (isTablePixelsForced(editor)) {
+    return { ...defaultAttributes, ...Obj.filter(defaultStyles, (_value, key) => Arr.contains(stylesAvailableAsAttributes, key)), width: getPixelForcedWidth(editor) };
+  } else {
+    return { ...defaultAttributes, ...Obj.filter(defaultStyles, (_value, key) => Arr.contains(stylesAvailableAsAttributes, key)) };
+  }
+};
```

Comment:
This should definitely not be looking at the `table_default_styles` option... we should just be setting the width in the attribute instead and not setting it in the default styles since it cannot be a fixed value due to the `table_style_by_css` option.

- [ ] Remove `width: '100%'` from the defaults for `table_default_styles`
- [ ] Change this to set the default width as a variable and use it when in the `determineDefault...` functions

Reply:
To be clear, I'd expect something like this:
```ts
const defaultWidth = '100%';

const getPixelForcedWidth = (editor: Editor) => {
  // Determine the inner size of the parent block element where the table will be inserted
  const dom = editor.dom;
  const parentBlock = dom.getParent<HTMLElement>(editor.selection.getStart(), dom.isBlock) ?? editor.getBody();
  return Width.getInner(SugarElement.fromDom(parentBlock)) + 'px';
};

// Note: This is also contained in the table plugin Options.ts file
const determineDefaultTableStyles = (editor: Editor, defaultStyles: Record<string, string>): Record<string, string> => {
  if (isTablePercentagesForced(editor)) {
    return { ...defaultStyles, width: defaultWidth };
  } else if (isTablePixelsForced(editor)) {
    return { ...defaultStyles, width: getPixelForcedWidth(editor) };
  } else {
    return defaultStyles;
  }
};

// Note: This is also contained in the table plugin Options.ts file
const determineDefaultTableAttributes = (editor: Editor, defaultAttributes: Record<string, string>): Record<string, string> => {
  if (isTablePercentagesForced(editor)) {
    return { ...defaultAttributes, width: defaultWidth };
  } else if (isTablePixelsForced(editor)) {
    return { ...defaultAttributes, width: getPixelForcedWidth(editor) };
  } else {
    return defaultAttributes;
  }
};
```

Note: In the above I did intentionally swap to using `isTablePercentagesForced` as it's easy to reason what's going on and makes sure we only add the width when it needs to be added.

Reply:
Oops the above doesn't account for the `table_style_by_css` option sorry... let me have a bit more of a think then.

Reply:
I thought it'd just be easier to do locally, so have made the changes mentioned above and pushed them here so you can review: https://github.com/tinymce/tinymce/compare/feature/TINY-8758...feature/TINY-8758_2. If all looks good, feel free to just cherry-pick that across.

The other thing missed was that we needed to move the registration of the `table_style_by_css` option to the model from the plugin.

Reply:
It was an attempt to have it in the css and moved over to the attributes when needed, and to allow more attributes to be moved over when appropriate instead of hard-coding them like this. I suppose it might have been a bit over-ambitious.